### PR TITLE
Change code coverage related build option for GCC

### PR DESCRIPTION
**Pull request by Wenxing-hou**
REF: #157 
Remove code coverage related build option for GCC by default.
Add GCOV as an option can be selected, and default is OFF.

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

### CODE REVIEW

**Comment by jyao1** [comment]
Can you please also update document to describe how to enable code coverage ?

**Comment by Wenxing-hou** [comment]
OK.  I will update the GCOV option document.


wenxing

**Comment by Wenxing-hou** [comment]
OK. I have updated the GCOV option document.

**Comment by jyao1** [comment]
I hardly understand the readme. Please

1) remove the change the readme.md.

2) add a commend in test.md. Please use `PARAM` style.

**Comment by jyao1** [comment]
Please change "Build cases(Note: Code Coverage need to set the GCOV option=ON.)." to "Build cases with `-DGCOV=ON`."

**Comment by jyao1** [comment]
Please remove below. It is unncessary.
```
   When `-DGCOV` option is no input,such as:cmake -DARCH=x64 -DTOOLCHAIN=GCC -DTARGET=Debug -DCRYPTO=mbedtls ..,
   the GCOV is OFF by default.
```

**Comment by jyao1** [comment]
Looks good.

File: doc/test.md [hunk 1]
```diff
@@ -102,7 +102,17 @@ For riscv64: `qemu-riscv64 -L /usr/riscv64-linux-gnu <TestBinary>`
...
    Install lcov `sudo apt-get install lcov`.
 
-   Build cases.
+   Build cases with `-DGCOV=ON`.
+
+   ```
+   cd libspdm
+   mkdir build
+   cd build
+   cmake -DARCH=<x64|ia32|arm|aarch64|riscv32|riscv64|arc> -DTOOLCHAIN=GCC -DTARGET=<Debug|Release> -DCRYPTO=<mbedtls|openssl> -DGCOV=ON..
```
Comment (steven-bellock) [inline doc/test.md:111]: Need space after -DGCOV=ON
`-DGCOV=ON ..`

Comment (Wenxing-hou) [reply doc/test.md]: I have updated it.  I appreciate your advise!  Thanks!

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**